src/FileService.Data/FileRepository.cs

```diff
@@ -23,17 +23,12 @@ public FileRepository(IDataProvider provider, IHttpContextAccessor httpContextAc
       _httpContextAccessor = httpContextAccessor;
     }
 
-    public async Task<bool> CreateAsync(List<DbFile> files)
+    public async Task<List<DbFile>> CreateAsync(List<DbFile> files)
     {
-      if (files == null || !files.Any())
-      {
-        return false;
-      }
-
       _provider.Files.AddRange(files);
       await _provider.SaveAsync();
 
-      return true;
+      return files.Where(x => !_provider.Files.Select(f => f.Id).ToList().Contains(x.Id)).ToList();
```

Comment:
Why not return files.Where(x => !_provider.Files.Select(f => f.Id).Contains(x.Id)).ToList();?

Reply:
sure, thanks =)